### PR TITLE
feat: public profile connection, buyer page

### PR DESCRIPTION
**Pull request by Blaxwo**
## Describe your changes

- I created connection for backend and fronted for public profile page.
- I tested redux for my pages.
- I created new page "Buyer profile page"

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-157?atlOrigin=eyJpIjoiY2ExNWZmOTMxNWQ4NGM1ODg0OTYwYzNiZTQwMDk4YzgiLCJwIjoiaiJ9)
- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-157?atlOrigin=eyJpIjoiMzc0YjJmNjA0YTllNDE2M2IyNzI3NzFjYjg3NzJkMmYiLCJwIjoiaiJ9)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
зображення

### CODE REVIEW

File: src/__tests__/reduxQueries.test.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import { waitFor } from '@testing-library/dom';
+import { act } from '@testing-library/react';
+import { renderHook } from '@testing-library/react-hooks';
+import { RTKUrls } from 'common/constants';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+
+import { IPublicUser, IReview } from '../redux/user/types';
```
Comment (lyudad) [inline src/__tests__/reduxQueries.test.ts:8]: alias not work?

File: src/pages/VendorProfilePage/index.tsx [hunk 2]
```diff
@@ -15,19 +28,61 @@ import {
...
+  }, [userError, reviewsError, showToast, t]);
+
+  if (!user || !reviewsData) {
+    return <CircularProgress />;
+  }
+
+  const userName = user.name || '';
+  const userAvatar = user.photoUrl || '';
+  const userRating = user.rating || 0;
+  const reviewCount = reviewsData.length || 0;
```
Comment (lyudad) [inline src/pages/VendorProfilePage/index.tsx]: у тебя это повторяется, может вынести в функцию что бы не дублировать?

Comment (Blaxwo) [reply src/pages/VendorProfilePage/index.tsx]: Согласна, я вынесла это в отдельную функцию и теперь использую на двух страницах

File: src/__tests__/reduxQueries.test.ts (hunk 1 above)
Comment (Blaxwo) [reply src/__tests__/reduxQueries.test.ts:8]: My mistake, I added alias

**Pull request merged**